src/main/java/com/bigchaindb/builders/BigchainDbTransactionBuilder.java

```diff
@@ -326,11 +326,15 @@ public IBuild build(EdDSAPublicKey publicKey) {
                 this.transaction.addInput(input);
             }
 
-            if (this.transaction.getOperation() == null) {
+            if (this.operation == null || this.operation == Operations.CREATE) {
                 this.transaction.setOperation("CREATE");
+                this.transaction.setAsset(new Asset(this.assets, this.assetsDataClass));
+            }
+            if (this.operation == Operations.TRANSFER){
+                this.transaction.setOperation(String.valueOf(this.operation));
+                this.transaction.setAsset(new Asset(this.inputs.get(0).getFulFills().getTransactionId()));
```

Comment:
This is not correct. This will only work in case of CREATE to TRANSFER transactions. Not in case of TRANSFER to TRANSFER transactions. In T -> T tx, we still need Asset Id, but id in fulfils object is not same Asset id but transaction id of previous transfer.